server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreIT.java

```diff
@@ -277,4 +278,35 @@ public void testRemoteSegmentCleanup() throws Exception {
     public void testRemoteTranslogCleanup() throws Exception {
         verifyRemoteStoreCleanup(true);
     }
+
+    public void testStaleCommitDeletion() throws Exception {
+        internalCluster().startDataOnlyNodes(3);
+        createIndex(INDEX_NAME, remoteStoreIndexSettings(1, 10000));
+        int numberOfIterations = randomIntBetween(5, 15);
+        boolean invokeFlush = randomBoolean();
```

Comment:
Why do we need to take random boolean here? For deletion to happen, commit is required. Do we really need to test for no commit cases?

Reply:
In that case, we should create 2 tests. The problem with `randomBoolean()` is it has a potential to create flaky test. If we test something in local that goes through one branch and passes does not given confidence that it will pass for all the branches.